### ISSUE TITLE
Cannot connect to Playwright server

### ISSUE THREAD

**Issue by alxppp** [YusukeIwaki/playwright-ruby-client#254]
### Usecase / What to achieve
Connect client to a Playwright server on the same Docker network.

### Question

I'm running into the following error when connecting to a Playwright server:
```rb
irb(main):012:1* Playwright.connect_to_browser_server('ws://playwright:8080') do |browser|
irb(main):013:1*   page = browser.new_page
irb(main):014:1*   page.goto('https://github.com/YusukeIwaki')
irb(main):015:1*   page.screenshot(path: 'YusukeIwaki.png')
irb(main):016:0> end
Traceback (most recent call last):
       11: from /usr/local/bin/irb:23:in `<main>'
       10: from /usr/local/bin/irb:23:in `load'
        9: from /usr/local/lib/ruby/gems/2.7.0/gems/irb-1.2.6/exe/irb:11:in `<top (required)>'
        8: from (irb):11
        7: from (irb):12:in `rescue in irb_binding'
        6: from /usr/local/bundle/gems/playwright-ruby-client-1.34.0/lib/playwright.rb:160:in `connect_to_browser_server'
        5: from /usr/local/bundle/gems/playwright-ruby-client-1.34.0/lib/playwright/connection.rb:50:in `initialize_playwright'
        4: from /usr/local/bundle/gems/playwright-ruby-client-1.34.0/lib/playwright/connection.rb:104:in `send_message_to_server'
        3: from /usr/local/bundle/gems/concurrent-ruby-1.2.2/lib/concurrent-ruby/concurrent/promises.rb:1472:in `value!'
        2: from /usr/local/bundle/gems/concurrent-ruby-1.2.2/lib/concurrent-ruby/concurrent/promises.rb:1258:in `wait_until_resolved!'
        1: from /usr/local/bundle/gems/concurrent-ruby-1.2.2/lib/concurrent-ruby/concurrent/promises.rb:1258:in `raise'
Playwright::Error (TypeError: Cannot read properties of undefined (reading 'launch'))
```

Same error when using the `connect_to_playwright_server` method:
```rb
irb(main):009:1* Playwright.connect_to_playwright_server('ws://playwright:8080') do |playwright|
irb(main):010:2*   playwright.chromium.launch do |browser|
irb(main):011:2*     page = browser.new_page
irb(main):012:2*     page.goto('https://github.com/YusukeIwaki')
irb(main):013:2*     page.screenshot(path: './YusukeIwaki.png')
irb(main):014:1*   end
irb(main):015:0> end
Traceback (most recent call last):
       10: from /usr/local/bin/irb:23:in `<main>'
        9: from /usr/local/bin/irb:23:in `load'
        8: from /usr/local/lib/ruby/gems/2.7.0/gems/irb-1.2.6/exe/irb:11:in `<top (required)>'
        7: from (irb):9
        6: from /usr/local/bundle/gems/playwright-ruby-client-1.34.0/lib/playwright.rb:123:in `connect_to_playwright_server'
        5: from /usr/local/bundle/gems/playwright-ruby-client-1.34.0/lib/playwright/connection.rb:50:in `initialize_playwright'
        4: from /usr/local/bundle/gems/playwright-ruby-client-1.34.0/lib/playwright/connection.rb:104:in `send_message_to_server'
        3: from /usr/local/bundle/gems/concurrent-ruby-1.2.2/lib/concurrent-ruby/concurrent/promises.rb:1472:in `value!'
        2: from /usr/local/bundle/gems/concurrent-ruby-1.2.2/lib/concurrent-ruby/concurrent/promises.rb:1258:in `wait_until_resolved!'
        1: from /usr/local/bundle/gems/concurrent-ruby-1.2.2/lib/concurrent-ruby/concurrent/promises.rb:1258:in `raise'
Playwright::Error (TypeError: Cannot read properties of undefined (reading 'launch'))
```

---

It doesn't seem to be the connection that's not working, when I turn off the Playwright server, I'm getting a different error:
```rb
Traceback (most recent call last):
        9: from /usr/local/bin/irb:23:in `<main>'
        8: from /usr/local/bin/irb:23:in `load'
        7: from /usr/local/lib/ruby/gems/2.7.0/gems/irb-1.2.6/exe/irb:11:in `<top (required)>'
        6: from (irb):16
        5: from (irb):17:in `rescue in irb_binding'
        4: from /usr/local/bundle/gems/playwright-ruby-client-1.34.0/lib/playwright.rb:156:in `connect_to_browser_server'
        3: from /usr/local/bundle/gems/playwright-ruby-client-1.34.0/lib/playwright/connection.rb:39:in `async_run'
        2: from /usr/local/bundle/gems/playwright-ruby-client-1.34.0/lib/playwright/web_socket_transport.rb:58:in `async_run'
        1: from /usr/local/bundle/gems/playwright-ruby-client-1.34.0/lib/playwright/web_socket_transport.rb:84:in `rescue in async_run'
Playwright::WebSocketClient::TransportError (Connection refused - connect(2) for "playwright" port 8080)
```

**Comment by YusukeIwaki** [comment]
It seems the specification of `playwright run-server` was changed. I will check it.

**Comment by YusukeIwaki** [comment]
@alxppp It seems `export PW_EXTENSION_MODE=1` would enable playwright server feature.
https://github.com/microsoft/playwright/pull/23215

However playwright server feature is just an internal feature of Playwright test runner and can be changed unexpectedly.
Browser server is a good alternative.

```
npx playwright@1.35.0 launch-server --browser chromium 
```

However playwright-ruby-client 1.34 is buggy, it will be fixed in 1.35. Sorry for inconvenience.

**Comment by alxppp** [comment]
Thanks a lot for the quick response @YusukeIwaki.

> Please keep playwright driver version at 1.34.x at this moment by explicitly specifying the version playwright@1.34.3. 

Unfortunately, I get the same error on `1.34.3`.

But I switched to browser server after you published the `1.35` gem version, and everything is working again – thank you!
